crates/bevy_sprite/src/tilemap_chunk/mod.rs

```diff
@@ -0,0 +1,263 @@
+        }
+
+        let Some(material) = materials.get_mut(material.id()) else {
+            warn!(
+                "TilemapChunkMaterial not found for tilemap chunk {}",
+                chunk_entity
+            );
+            continue;
+        };
+        let Some(indices_image) = images.get_mut(&material.indices) else {
+            warn!(
+                "TilemapChunkMaterial indices image not found for tilemap chunk {}",
+                chunk_entity
+            );
+            continue;
+        };
+        let Some(data) = indices_image.data.as_mut() else {
+            warn!(
+                "TilemapChunkMaterial indices image data not found for tilemap chunk {}",
+                chunk_entity
+            );
+            continue;
+        };
+        data.clear();
+        data.extend(
+            indices
+                .iter()
+                .copied()
+                .flat_map(|i| u16::to_ne_bytes(i.unwrap_or(u16::MAX))),
+        );
+    }
+}
+
+fn make_chunk_image(size: &UVec2, indices: &[Option<u16>]) -> Image {
+    Image {
+        data: Some(
+            indices
+                .iter()
+                .copied()
+                .flat_map(|i| u16::to_ne_bytes(i.unwrap_or(u16::MAX)))
+                .collect(),
+        ),
+        texture_descriptor: TextureDescriptor {
+            size: Extent3d {
+                width: size.x,
+                height: size.y,
+                depth_or_array_layers: 1,
+            },
+            dimension: TextureDimension::D2,
+            format: TextureFormat::R16Uint,
+            label: None,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING | TextureUsages::COPY_DST,
+            view_formats: &[],
+        },
+        sampler: ImageSampler::nearest(),
+        texture_view_descriptor: None,
+        asset_usage: RenderAssetUsages::RENDER_WORLD | RenderAssetUsages::MAIN_WORLD,
+    }
+}
+
+fn make_chunk_mesh(size: &UVec2, display_size: &Vec2, anchor: &Anchor) -> Mesh {
+    let mut mesh = Mesh::new(
+        PrimitiveTopology::TriangleList,
+        RenderAssetUsages::RENDER_WORLD | RenderAssetUsages::MAIN_WORLD,
+    );
+
+    let offset = display_size * (Vec2::splat(-0.5) - anchor.as_vec());
+
+    let num_quads = size.element_product() as usize;
+    let quad_size = display_size / size.as_vec2();
+
+    let mut positions = Vec::with_capacity(4 * num_quads);
+    let mut uvs = Vec::with_capacity(4 * num_quads);
+    let mut indices = Vec::with_capacity(6 * num_quads);
+
+    for y in 0..size.y {
+        for x in 0..size.x {
+            let i = positions.len() as u32;
+
+            let p0 = offset + quad_size * UVec2::new(x, y).as_vec2();
+            let p1 = p0 + quad_size;
+
+            positions.extend([
+                Vec3::new(p0.x, p0.y, 0.0),
+                Vec3::new(p1.x, p0.y, 0.0),
+                Vec3::new(p0.x, p1.y, 0.0),
+                Vec3::new(p1.x, p1.y, 0.0),
```

Comment:
this is duplicating all the vertex positions (except the outside border). not a huge deal but since we're indexing anyway it would be nice to avoid it.

Reply:
We need it because we're using vertex index primitive in the shader divided by 4 to get the tile index, so every tile needs 4 vertices. I think the chunk should just be one quad personally but we can wait on that and benchmark compare later.

Reply:
... or you use a uv that's 0..1 over the whole chunk, and imply the tile from that. but anyway, not important enough to block on imo